extensions/ql-vscode/src/helpers.ts

```diff
@@ -737,3 +742,20 @@ export async function* walkDirectory(
     }
   }
 }
+
+export const QLPACK_FILENAMES = ["qlpack.yml", "codeql-pack.yml"];
+export const FALLBACK_QLPACK_FILENAME = QLPACK_FILENAMES[0];
+
+export async function getQlPackPath(
+  packRoot: string,
+): Promise<string | undefined> {
+  for (const filename of QLPACK_FILENAMES) {
+    const path = join(packRoot, filename);
+
+    if (await pathExists(path)) {
+      return path;
+    }
+  }
+
+  return undefined;
```

Comment:
Could we have a test for this new function?

Reply:
👍 , I've added a test for this function. I've also moved the function to the `pure` directory because this doesn't actually need VSCode and that makes it clearer that this can be a unit test.